zwave_js_server/model/node/__init__.py

```diff
@@ -463,11 +465,28 @@ async def async_send_command(
         if require_schema is not None:
             kwargs["require_schema"] = require_schema
 
-        if wait_for_result or (
-            wait_for_result is None and self.status != NodeStatus.ASLEEP
-        ):
+        if wait_for_result:
             result = await self.client.async_send_command(message, **kwargs)
             return result
+        if wait_for_result is None and self.status not in (
+            NodeStatus.ASLEEP,
+            NodeStatus.DEAD,
+        ):
+            result_task = asyncio.create_task(
+                self.client.async_send_command(message, **kwargs)
+            )
+            status_task = asyncio.create_task(
+                self._status_event.wait()
+            )
+            await asyncio.wait(
+                [result_task, status_task],
+                return_when=asyncio.FIRST_COMPLETED,
+            )
+            status_task.cancel()
```

Comment:
OK so `asyncio.wait` wants Tasks, not coroutine. So now I wrapped the status event wait coro in a Task. It's appropriate to cancel the task here right?

Reply:
I think so.